frontend/src/app/main/ui/export.cljs

```diff
@@ -0,0 +1,211 @@
+;; This Source Code Form is subject to the terms of the Mozilla Public
+;; License, v. 2.0. If a copy of the MPL was not distributed with this
+;; file, You can obtain one at http://mozilla.org/MPL/2.0/.
+;;
+;; Copyright (c) UXBOX Labs SL
+
+(ns app.main.ui.export
+  "Assets exportation common components."
+  (:require
+   [app.common.data :as d]
+   [app.common.data.macros :as dm]
+   [app.main.data.exports :as de]
+   [app.main.data.modal :as modal]
+   [app.main.refs :as refs]
+   [app.main.store :as st]
+   [app.main.ui.icons :as i]
+   [app.main.ui.workspace.shapes :refer [shape-wrapper]]
+   [app.util.dom :as dom]
+   [app.util.i18n :as i18n :refer  [tr c]]
+   [cuerdas.core :as str]
+   [rumext.alpha :as mf]))
+
+(mf/defc export-shapes-dialog
+  {::mf/register modal/components
+   ::mf/register-as :export-shapes}
+  [{:keys [exports filename]}]
+  (let [lstate          (mf/deref refs/export)
+        in-progress?    (:in-progress lstate)
+
+        exports         (mf/use-state exports)
+
+        all-exports     (deref exports)
+        all-checked?    (every? :enabled all-exports)
+        all-unchecked?  (every? (complement :enabled) all-exports)
+
+        enabled-exports (into [] (filter :enabled) @exports)
+
+        cancel-fn
+        (fn [event]
```

Comment:
These functions `cancel-fn`, `accept-fn`... should use `mf/use-callback`

Reply:
On the modals it is not necesarry because they are not rerendered all the time, so it is just simplier not using this. As we already do on the rest of penpot codebase (not all callbacks uses use-callback right now).